core/variables.js

```diff
@@ -123,12 +120,15 @@ Blockly.Variables.flyoutCategory = function(workspace) {
 
   xmlList.push(button);
 
-  if (variableNameList.length > 0) {
+  if (variableModelList.length > 0) {
+    var firstVariable = variableModelList[0];
     if (Blockly.Blocks['variables_set']) {
       var gap = Blockly.Blocks['math_change'] ? 8 : 24;
       var blockText = '<xml>' +
             '<block type="variables_set" gap="' + gap + '">' +
-            '<field name="VAR">' + variableNameList[0] + '</field>' +
+            '<field name="VAR" variableType="' + firstVariable.type +
```

Comment:
Make a helper function to generate these field strings (multiple lines, repeated multiple times, and easy to accidentally break).

Pass in a VariableModel and return a string.